src/app/zap-templates/templates/app/im-cluster-command-handler.zapt

```diff
@@ -91,8 +91,17 @@ void DispatchSingleClusterCommand(const ConcreteCommandPath & aCommandPath, TLV:
     {{/unless}}
     {{/all_user_clusters_with_incoming_commands}}
     default:
-        ChipLogError(Zcl, "Unknown cluster " ChipLogFormatMEI, ChipLogValueMEI(aCommandPath.mClusterId));
-        errorStatus = Protocols::InteractionModel::Status::UnsupportedCluster;
+        // If we already validated that the cluster does indeed exist (e.g. when dispatching to CommandHandlerInterface for that cluster).
+        if (aClusterIsValid)
+        {
+            errorStatus = Protocols::InteractionModel::Status::UnsupportedCommand;
+        }
+        else
+        {
+            ChipLogError(Zcl, "Unknown cluster " ChipLogFormatMEI, ChipLogValueMEI(aCommandPath.mClusterId));
+            errorStatus = Protocols::InteractionModel::Status::UnsupportedCluster;            
```

Comment:
How do you know it's not UnsupportedEndpoint?

Again, test coverage would flush out these edge cases...

Reply:
This is/was there before though...
I think codegen validates endpoint (I think ... did not check as we inherited this).

We should have integration tests that validate return codes on bad commands. If we do not we should be able to add some.

Reply:
https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TestInvokeReturnCodes.py